R/sidebar.R

```diff
@@ -112,16 +156,30 @@ layout_sidebar <- function(sidebar = sidebar(), ..., fillable = FALSE, fill = TR
     columns_collapse <- rev(columns_collapse)
   }
 
+  sidebar_max_height_mobile <-
+    if (is.null(sidebar$max_height_mobile)) {
+      if (!is.null(height)) {
+        "fit-content(50%)"
+      } else {
+        "fit-content(250px)"
+      }
+    } else {
+      sprintf("fit-content(%s)", sidebar$max_height_mobile)
+    }
+
   res <- div(
     class = "bslib-sidebar-layout",
     class = if (right) "sidebar-right",
-    class = if (isFALSE(sidebar$open)) "sidebar-collapsed",
+    class = if (identical(sidebar$open, "closed")) "sidebar-collapsed",
+    `data-sidebar-collapse-at-breakpoint` =
+      if (identical(sidebar$open, "desktop")) "true",
```

Comment:
I think it would make sense to provide the relevant CSS var in the inline style (instead of via this data attribute)?

Also, I think we need to _always_ provide a value for the CSS variable, since we don't want nested sidebars to inherit from their parent

Reply:
In the end:

* The R side adds a `data-sidebar-init-auto-collapse` attribute.
* The CSS sets `--bslib-sidebar-js-init-collapsed` at or below `breakpoint-sm`.
* The initialization JS auto inits sidebars matching the attribute, choosing to collapse them if the CSS variable is `"true"`.